python/example_code/ec2/scenario_get_started_instances.py

```diff
@@ -1,274 +1,334 @@
+
+                public_ip = instance.get("PublicIpAddress")
+                if public_ip:
+                    console.print(
+                        f"\tssh -i {self.key_wrapper.key_file_path} ec2-user@{public_ip}"
+                    )
+                else:
+                    console.print(
+                        "Instance does not have a public IP address assigned.",
+                        style="bold red",
+                    )
+            else:
+                console.print(
+                    "No instance available to retrieve public IP address.",
+                    style="bold red",
+                )
         else:
-            print(
-                f"\tssh -i {self.key_wrapper.key_file_path} "
-                f"ec2-user@{self.eip_wrapper.elastic_ip.public_ip}"
+            elastic_ip = self.eip_wrapper.elastic_ips[0]
+            response = self.eip_wrapper.ec2_client.describe_addresses(
+                AllocationIds=[elastic_ip.allocation_id]
+            )
+            elastic_ip_address = response["Addresses"][0]["PublicIp"]
+            console.print(
+                f"\tssh -i {self.key_wrapper.key_file_path} ec2-user@{elastic_ip_address}"
             )
-        q.ask("Press Enter when you're ready to continue the demo.")
 
-    @demo_func
     def associate_elastic_ip(self):
-        """
-        1. Allocates an Elastic IP address and associates it with the instance.
-        2. Displays an SSH connection string that uses the Elastic IP address.
-        """
-        print(
+        console.print("\n**Step 4: Allocate an Elastic IP Address**", style="bold cyan")
+        console.print(
             "You can allocate an Elastic IP address and associate it with your instance\n"
             "to keep a consistent IP address even when your instance restarts."
         )
-        elastic_ip = self.eip_wrapper.allocate()
-        print(f"Allocated static Elastic IP address: {elastic_ip.public_ip}.")
-        self.eip_wrapper.associate(self.inst_wrapper.instance)
-        print(f"Associated your Elastic IP with your instance.")
-        print(
+
+        with alive_bar(1, title="Allocating Elastic IP") as bar:
+            elastic_ip = self.eip_wrapper.allocate()
+            time.sleep(0.5)
+            bar()
+
+        console.print(
+            f"- **Allocated Static Elastic IP Address**: {elastic_ip.public_ip}."
+        )
+
+        with alive_bar(1, title="Associating Elastic IP") as bar:
+            self.eip_wrapper.associate(
+                elastic_ip.allocation_id, self.inst_wrapper.instances[0]["InstanceId"]
+            )
+            time.sleep(2)
+            bar()
+
+        console.print(f"- **Associated Elastic IP with Your Instance**.")
+        console.print(
             "You can now use SSH to connect to your instance by using the Elastic IP."
         )
         self._display_ssh_info()
 
-    @demo_func
     def stop_and_start_instance(self):
-        """
```

Comment:
Totally agree on commenting out functions that execute SDK code. However, the scenario functions here are already explained in the print statements and I don't see a ton of value also adding a text block explaining them. I'm happy to add this back if you feel strongly because it doesn't hurt, but it just feels excessive and like more stuff we need to maintain in terms of keeping it accurate over time.

Reply:
I think they add value when viewed in the docs, since at that point the customer is reading the code, not necessarily running it.

Reply:
I disagree, but I think you have the better articulated argument so I'll add these back for all the functions in this scenario file.